samples/extensions/shader_quad_control/shader_quad_control.cpp

```diff
@@ -0,0 +1,190 @@
+ * limitations under the License.
+ */
+
+#include "shader_quad_control.h"
+
+ShaderQuadControl::ShaderQuadControl()
+{
+	title = "Shader quad control";
+	set_api_version(VK_API_VERSION_1_2);
+
+	add_instance_extension(VK_KHR_GET_PHYSICAL_DEVICE_PROPERTIES_2_EXTENSION_NAME);
+	// VK_KHR_shader_quad_control requires VK_KHR_shader_maximal_reconvergence per spec
+	add_device_extension(VK_KHR_SHADER_MAXIMAL_RECONVERGENCE_EXTENSION_NAME);
+	add_device_extension(VK_KHR_SHADER_QUAD_CONTROL_EXTENSION_NAME);
+}
+
+ShaderQuadControl::~ShaderQuadControl()
+{
+	if (has_device())
+	{
+		if (pipeline != VK_NULL_HANDLE)
+		{
+			vkDestroyPipeline(get_device().get_handle(), pipeline, nullptr);
+		}
+		if (pipeline_layout != VK_NULL_HANDLE)
+		{
+			vkDestroyPipelineLayout(get_device().get_handle(), pipeline_layout, nullptr);
+		}
+	}
+}
+
+bool ShaderQuadControl::prepare(const vkb::ApplicationOptions &options)
+{
+	if (!ApiVulkanSample::prepare(options))
+	{
+		return false;
+	}
+
+	create_pipeline_layout();
+	create_pipeline();
+	build_command_buffers();
+
+	prepared = true;
+	return true;
+}
+
+void ShaderQuadControl::request_gpu_features(vkb::core::PhysicalDeviceC &gpu)
+{
+	REQUEST_REQUIRED_FEATURE(gpu, VkPhysicalDeviceShaderQuadControlFeaturesKHR, shaderQuadControl);
+}
+
+void ShaderQuadControl::create_pipeline_layout()
+{
+	VkPipelineLayoutCreateInfo pipeline_layout_ci{VK_STRUCTURE_TYPE_PIPELINE_LAYOUT_CREATE_INFO};
+	VK_CHECK(vkCreatePipelineLayout(get_device().get_handle(), &pipeline_layout_ci, nullptr, &pipeline_layout));
+}
+
+void ShaderQuadControl::create_pipeline()
+{
+	std::array<VkPipelineShaderStageCreateInfo, 2> stages{};
+	stages[0] = load_shader("shader_quad_control/glsl/quad_control.vert.spv", VK_SHADER_STAGE_VERTEX_BIT);
+	stages[1] = load_shader("shader_quad_control/glsl/quad_control.frag.spv", VK_SHADER_STAGE_FRAGMENT_BIT);
+
+	VkPipelineVertexInputStateCreateInfo vertex_input{VK_STRUCTURE_TYPE_PIPELINE_VERTEX_INPUT_STATE_CREATE_INFO};
+
+	VkPipelineInputAssemblyStateCreateInfo input_assembly{VK_STRUCTURE_TYPE_PIPELINE_INPUT_ASSEMBLY_STATE_CREATE_INFO};
+	input_assembly.topology = VK_PRIMITIVE_TOPOLOGY_TRIANGLE_LIST;
+
+	VkPipelineViewportStateCreateInfo viewport_state{VK_STRUCTURE_TYPE_PIPELINE_VIEWPORT_STATE_CREATE_INFO};
+	viewport_state.viewportCount = 1;
+	viewport_state.scissorCount  = 1;
+
+	VkPipelineRasterizationStateCreateInfo raster{VK_STRUCTURE_TYPE_PIPELINE_RASTERIZATION_STATE_CREATE_INFO};
+	raster.polygonMode = VK_POLYGON_MODE_FILL;
+	raster.cullMode    = VK_CULL_MODE_NONE;
+	raster.frontFace   = VK_FRONT_FACE_COUNTER_CLOCKWISE;
+	raster.lineWidth   = 1.0f;
+
+	VkPipelineMultisampleStateCreateInfo msaa{VK_STRUCTURE_TYPE_PIPELINE_MULTISAMPLE_STATE_CREATE_INFO};
+	msaa.rasterizationSamples = VK_SAMPLE_COUNT_1_BIT;
+
+	VkPipelineDepthStencilStateCreateInfo depth_stencil{VK_STRUCTURE_TYPE_PIPELINE_DEPTH_STENCIL_STATE_CREATE_INFO};
+	depth_stencil.depthTestEnable  = VK_FALSE;
+	depth_stencil.depthWriteEnable = VK_FALSE;
```

Comment:
Why do you emphasize that no depth test or writing is done here?
You actually don't **need** to explicitly set those members to false.